static/js/disaster_dashboard/map_section.tsx

```diff
@@ -0,0 +1,270 @@
+    ReactDOM.render(
+      <InfoCard
+        onClose={() =>
+          d3.select(infoCardRef.current).style("visibility", "hidden")
+        }
+        eventData={point}
+      />,
+      infoCardRef.current
+    );
+    // get info card dimensions
+    const infoCardRect = (
+      infoCard.node() as HTMLDivElement
+    ).getBoundingClientRect();
+    const infoCardHeight = infoCardRect.height;
+    const infoCardWidth = infoCardRect.width;
+    const containerWidth = (
+      d3.select(svgContainer.current).node() as HTMLDivElement
+    ).getBoundingClientRect().width;
+    // calculate left and top position to show the info card
+    let left = Math.min(event.offsetX, containerWidth - infoCardWidth);
+    if (left < 0) {
+      left = 0;
+      infoCard.style("width", containerWidth + "px");
+    } else {
+      infoCard.style("width", "fit-content");
+    }
+    let top = event.offsetY - infoCardHeight - iNFO_CARD_OFFSET;
+    if (top < 0) {
+      top = event.offsetY + iNFO_CARD_OFFSET;
+    }
+    // set info card position and make it visible
+    infoCard
+      .style("left", left + "px")
+      .style("top", top + "px")
+      .style("visibility", "visible");
+  }
+
+  function canClickRegion(placeDcid: string): boolean {
+    // If a country is a European country, we want to add Europe to its list
+    // of parents. This is important for getAllChildPlaceTypes (line 191)
+    // because it will handle European countries differently than regular
+    // countries.
+    const enclosingPlace =
+      props.fetchedEuropeanPlaces.findIndex(
+        (country) => country.dcid === placeDcid
+      ) > -1
+        ? EUROPE_NAMED_TYPED_PLACE
+        : props.selectedPlaceInfo.place;
+    const parentPlaces = getParentPlaces(
+      props.selectedPlaceInfo.place,
+      enclosingPlace,
+      props.selectedPlaceInfo.parentPlaces
+    );
+    const newPlaceAsNamedTypedPlace = {
+      dcid: placeDcid,
+      name: placeDcid,
+      types: [props.selectedPlaceInfo.placeType],
+    };
+    return !_.isEmpty(
+      getAllChildPlaceTypes(newPlaceAsNamedTypedPlace, parentPlaces).filter(
+        (placeType) =>
+          placeType !== IPCC_PLACE_50_TYPE_DCID &&
+          placeType !== props.selectedPlaceInfo.placeType
+      )
+    );
+  }
+
+  function draw(): void {
+    const width = svgContainer.current.offsetWidth;
+    const height = (width * 2) / 5;
+    const zoomParams = {
+      zoomInButtonId: ZOOM_IN_BUTTON_ID,
+      zoomOutButtonId: ZOOM_OUT_BUTTON_ID,
+    };
+    document.getElementById(MAP_ID).innerHTML = "";
+    const isUsaPlace = isChildPlaceOf(
+      props.selectedPlaceInfo.place.dcid,
+      USA_PLACE_DCID,
+      props.selectedPlaceInfo.parentPlaces
+    );
+    const projection = getProjection(isUsaPlace, "", width, height);
+    drawD3Map(
+      MAP_ID,
+      props.geoJson,
+      height,
+      width,
+      {} /** dataValues: no data values to show on the base map */,
```

Comment:
super nit: only one star /* ... */ if it's not multiline:
https://google.github.io/styleguide/tsguide.html#comments-when-calling-a-function

thanks for adding this! i also prefer the comment after the argument (since it's not prescribed by the style guide).

Reply:
done